### PR TITLE
[Balance Druid] Moons casts added

### PR DESCRIPTION
**Pull request by Iskalla**
The code on every one of the modules is pretty similar :)

### CODE REVIEW

File: src/Parser/BalanceDruid/Modules/Spells/FullMoon.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    orderFound = false;
+
+    on_byPlayer_cast(event) {
+        if (!this.firstCast) {
+            this.firstCastTime = event.timestamp;
+            this.firstCast = true;
+        }
+
+        const spellId = event.ability.guid;
+        if (this.orderFound || (SPELLS.FULL_MOON.id !== spellId && SPELLS.HALF_MOON.id !== spellId && SPELLS.NEW_MOON.id !== spellId)) {
```
Comment (rp4rk) [inline src/Parser/BalanceDruid/Modules/Spells/FullMoon.js]: Can we extract this logic out into a function, something like `isMoonSpell`?. Look at the Discipline `isAtonement` function for a similar premise.

File: src/Parser/BalanceDruid/Modules/Spells/FullMoon.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            icon={<SpellIcon id={SPELLS.FULL_MOON.id} />}
+            value={`${fmCasted}/${fmAvailableCasts}`}
+            label='Full Moon casts'
+        />
+        );
+    }
+    statisticOrder = STATISTIC_ORDER.CORE(5);
+}
+  
+export default FullMoon;
```
[the remark this answers is not in the document]
Comment (Iskalla) [reply src/Parser/BalanceDruid/Modules/Spells/FullMoon.js]: I add new lines but it refuses to realize it... xD

[the remark this answers is not in the document]
Comment (Iskalla) [reply src/Parser/BalanceDruid/Modules/Spells/FullMoon.js]: Should be fine now, seemed to be a windows problem

File: src/Parser/BalanceDruid/Modules/Spells/HalfMoon.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            icon={<SpellIcon id={SPELLS.HALF_MOON.id} />}
+            value={`${hmCasted}/${hmAvailableCasts}`}
+            label='Half Moon casts'
+        />
+        );
+    }
+    statisticOrder = STATISTIC_ORDER.CORE(4);
+}
+  
+export default HalfMoon;
```
[the remark this answers is not in the document]
Comment (Iskalla) [reply src/Parser/BalanceDruid/Modules/Spells/HalfMoon.js]: It's apparently fisex even tho it doesnt show as outdated :D

File: src/Parser/BalanceDruid/Modules/Spells/NewMoon.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+        const extraMoons = ((totalFromCD / 3) - eachMoon) * 3;
+        if (extraMoons > this.newMoonOrder) nmAvailableCasts ++;
+
+        const nmCasted = abilityTracker.getAbility(SPELLS.NEW_MOON.id).casts;
+        const percCasted = nmCasted / nmAvailableCasts;
+
+        when(percCasted).isLessThan(1)
+        .addSuggestion((suggest, actual, recommended) => {
+            return suggest(<span> Your <SpellLink id={SPELLS.NEW_MOON.id} /> cast efficiency can be improved, try maintaining yourself at low Moon charges at all times; you should (almost) never be at max (3) charges.</span>)
```
Comment (MartijnHols) [inline src/Parser/BalanceDruid/Modules/Spells/NewMoon.js]: `try maintaining yourself at low Moon charges at all times`

I had to read this a few times before I understood it, could you please change it a bit to make it clearer?

Comment (Iskalla) [reply src/Parser/BalanceDruid/Modules/Spells/NewMoon.js]: English not my main language :D
Is this enough? If not, can you provide an alternative, please? ^^u

Comment (MartijnHols) [reply src/Parser/BalanceDruid/Modules/Spells/NewMoon.js]: I suggested `Your New Moon cast efficiency can be improved. Try keeping at least 1 charge on cooldown; you should (almost) never be at max charges.` earlier in Discord. ;)

**Comment by MartijnHols** [comment]
Thanks!

**Pull request merged**